### PR TITLE
Add View Consultation Button

### LINKED ISSUE #9272: Add "View Latest Consultation" Button on the Patient Details Page
**Is your feature request related to a problem? Please describe.**

- [ ] Currently, on the patient details page, to check the live consultation, you need to click on the encounter in the nav bar.

**Steps to Replicate**

1. Go to the Patient tab.
2. Click on any live patient card.
3. In the patient consultation page, click on the Patient Details page in the top right corner.
4. View the card. The button should be implemented. 

**Expected Behavior**

- [ ] Add a "View Latest Encounter" button with the primary color and effects similar to other buttons in the card.
- [ ] This button should be visible only for live patients.

**Screenshot**

Image

### PR DESCRIPTION
**Pull request by JavidSumra**
## Proposed Changes

- Fixes #9272
- Added Button to View Latest Consultation

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
  - Introduced a label for accessing the latest patient encounter: "View Latest Encounter".
  - Added a new button component with authorization checks to enhance user interaction.

- **Bug Fixes**
  - Improved conditional rendering for consultation buttons based on patient data.

- **Refactor**
  - Reorganized component structure and logic for better readability and maintainability.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Let's also change all the ButtonV2's to use shadcn's Button instead in this page

File: src/components/Patient/PatientHome.tsx [hunk 1]
```diff
@@ -239,28 +239,48 @@ export const PatientHome = (props: {
                   <div className="space-y-3 border-b border-dashed text-left text-lg font-semibold text-secondary-900">
                     <div>
                       {patientData?.is_active &&
```
Comment (Jacobjeevan) [inline src/components/Patient/PatientHome.tsx]: Let's add a check for `facilityId === patientData.facility_object?.id.toString()` for the whole thing (display either button only if this is true).

File: src/components/Patient/PatientHome.tsx [hunk 2]
```diff
@@ -239,28 +239,48 @@ export const PatientHome = (props: {
...
+                        </div>
+                      ) : (
+                        <div>
+                          <ButtonV2
+                            className="w-full"
+                            size="default"
+                            onClick={() =>
+                              navigate(
+                                `/facility/${patientData.facility}/patient/${id}/consultation/${patientData.last_consultation?.id}/consent-records`,
+                              )
```
Comment (Jacobjeevan) [inline src/components/Patient/PatientHome.tsx]: remove `consent-records` from the url, encounter refers to consultation.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/ui/button.tsx [hunk 3]
```diff
@@ -16,7 +20,7 @@ const buttonVariants = cva(
         outline:
           "border border-gray-200 text-gray-900 bg-white shadow-sm hover:bg-gray-100 hover:text-gray-900 dark:border-gray-800 dark:bg-gray-950 dark:hover:bg-gray-800 dark:hover:text-gray-50",
         primary:
-          "bg-primary-700 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
+          "bg-primary-500 text-white shadow hover:bg-primary-500/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```
Comment (rithviknishad) [inline src/components/ui/button.tsx]: let's stick with whats as per new designs 

```suggestion
          "bg-primary-700 text-white shadow hover:bg-primary-700/90 dark:bg-primary-100 dark:text-primary-900 dark:hover:bg-primary-100/90",
```

File: src/components/ui/button.tsx [hunk 4]
```diff
@@ -60,4 +64,22 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
...
+const AuthorizedButton: React.FC<AuthorizedElementProps & ButtonProps> = ({
+  authorizeFor = () => true,
+  ...props
+}) => {
+  return (
+    <AuthorizedChild authorizeFor={authorizeFor}>
+      {({ isAuthorized }) => (
+        <Button
+          {...props}
+          disabled={props.disabled || !isAuthorized} // Disable if unauthorized or loading
```
Comment (rithviknishad) [inline src/components/ui/button.tsx]: code is self explanatory.
```suggestion
          disabled={props.disabled || !isAuthorized}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
lgtm, just some minor suggestions

File: src/components/Patient/PatientHome.tsx [hunk 5]
```diff
@@ -511,32 +534,30 @@ export const PatientHome = (props: {
                           <CareIcon icon="l-file-upload" className="text-xl" />
                           {t("view_update_patient_files")}
                         </span>
-                      </ButtonV2>
+                      </Button>
                     </div>
 
                     {NonReadOnlyUsers && (
```
Comment (rithviknishad) [inline src/components/Patient/PatientHome.tsx]: Although was there before, `NonReadOnlyUsers` is a function and boolean check would always return true right?

**Review by rithviknishad — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
image

@JavidSumra this button should be visible only for live patients, not discharged patients , do read the issue properly

**Comment by nihal467** [comment]
image

- [ ] The 'View Latest Consultation' button should appear only when the patient has an active consultation and When a patient is live with no consultation, the current behavior of showing 'Add Consultation' button should be retained

File: src/components/ui/button.tsx [hunk 6]
```diff
@@ -60,4 +64,19 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
...
+      {({ isAuthorized }) => (
+        <Button {...props} disabled={props.disabled || !isAuthorized}>
+          {props.children}
+        </Button>
+      )}
+    </AuthorizedChild>
+  );
+};
+
+export { Button, buttonVariants, AuthorizedButton };
```
Comment (rithviknishad) [inline src/components/ui/button.tsx]: can we move this to a separate file? let's not shadcn's files.

**Review by rithviknishad — COMMENTED** [review]
~by mistake~

File: src/components/ui/button.tsx (hunk 6 above)
Comment (JavidSumra) [reply src/components/ui/button.tsx]: Which file do you prefer @rithviknishad ?

Comment (rithviknishad) [reply src/components/ui/button.tsx]: src/components/common

File: package-lock.json
Comment (rithviknishad) [inline package-lock.json:1]: discard changes to the lockfile

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**